widgetssdk/src/main/java/com/glia/widgets/view/head/ActivityWatcherForChatHead.kt

```diff
@@ -38,15 +52,75 @@ internal class ActivityWatcherForChatHead(
 
     override fun onActivityPaused(activity: Activity) {
         controller.onActivityPaused()
+        removeChatHeadLayoutIfPresent()
         resumedActivity.clear()
     }
 
+    private fun createChatHeadLayout(activity: Activity) {
+        val chatHeadLayout = ChatHeadLayout(activity.baseContext)
+        chatHeadLayout.layoutParams = ConstraintLayout.LayoutParams(
+            ViewGroup.LayoutParams.MATCH_PARENT,
+            ViewGroup.LayoutParams.MATCH_PARENT
+        )
+        chatHeadLayout.setNavigationCallback(
+            object : ChatHeadLayout.NavigationCallback {
+                override fun onNavigateToEndScreenSharing() {
+                    navigateToEndScreenSharing(resumedActivity.get())
+                }
+
+                override fun onNavigateToChat() {
+                    navigateToChat(resumedActivity.get())
+                }
```

Comment:
I don't see a reason for the override function just calling private function. We could just remove private functions and put the logic in public override function

Reply:
This also applies for all other navigation functions

Reply:
@Pelkar Code is extracted to private functions just for readability. If we put everything to these override functions, the `setNavigationCallback` call will look like a mess to me, 3 lines of code will become 30+ lines of code.

I think that's not what we want, don't we?

Reply:
ok, fair enough.